### ISSUE TITLE
Role inheritance during item move

### ISSUE THREAD

**Issue by parphis** [DSpace/DSpace#3373]
**Describe the bug**
v6.3, XMLUI & v7.4 REST API
Moving an item from a collection to another with _Inherit policies_ checked does not apply the policies of the new collection for the item.

**To Reproduce**
 1. Create a collection where the items are open to read for Anonymous.
 2. Add another collection where the _Default read access_ has a group which contains the Administrators group.
 3. Move an item from the open access collection to the one created in the point 2 above.
 4. Check that the moved item is accessible by Anonymous even if the collection's default read is restricted.

**Expected behavior**
If an item - having open access rights previously - is moved to a restircted collection then it should not be available to everyone.

**Related work**
#3374

**Comment by peter975** [comment]
This is still happening in DSpace 7.4

**Comment by tdonohue** [comment]
@parphis or @peter975 : Could one of you port the existing PR #3374 to our `main` branch code? If so, I can try to find reviewers to see if we can get this fixed in 7.5.

**Comment by parphis** [comment]
@tdonohue I am happy to do it. I will open a PR for it.

**Comment by parphis** [comment]
@tdonohue I have checked the backend code regarding this issue. I found the followings:

 1. There is an if statement within the fn `move` https://github.com/DSpace/DSpace/blob/5c697c813bb62db13200f303291a4c92c00a441b/dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java#L988
We decide here whether to call the `adjustBundleBitstreamPolicies` and `adjustItemPolicies` functions later in order to inherit the destination collection's policy or not?
 2. The variable `inheritDefaultPolicies` never changes. It is always false at the backend even if I check the `Inherit policies` checkbox on the UI.
 3. The checkbox is working on the UI side well but as the comment writes there is no support for this inherit option on the backend side (PUT /api/core/items/<:uuid>/owningCollection): https://github.com/DSpace/dspace-angular/blob/0395480ab8ddeabe22ff1b7d2c6f678fd9830428/src/app/item-page/edit-item-page/item-move/item-move.component.ts#L29 

I have created a PR https://github.com/DSpace/DSpace/pull/8546 for the backend part and I have noticed the above things for the reviewers.
